doc/Route_priority_and_TTL.md

```diff
@@ -0,0 +1,131 @@
+"$edgeHub": {
+  "properties.desired": {
+    "routes": {
+      <b>"route1": {
+        "route": <string>,
+        "priority": <uint>,
+        "timeToLiveSecs": <uint>,
+        "Recency": {
+          "Seconds": <uint>,
+          "FallbackPriority": <uint>
+          }
+        },</b>
+      "route2": <string>
+    },
+    "storeAndForwardConfiguration": {
+      "timeToLiveSecs": 7200
+    }
+  }
+}
+</pre>
+
+### Example
+![](./images/route_priority_ttl/priority_ttl_example_deployment.jpg)
+
+<pre>
+"$edgeHub": {
+  "properties.desired": {
+    "schemaVersion": <b>"1.1.0",</b>
+    "routes": {
+      "secCamAlerts": {
+        "route": "FROM /messages/modules/SecCamMod/outputs/P0_Alerts INTO $upstream",
+        "priority": 0,
+        "timeToLiveSecs": 86400
+      },
+      "floorCamAlerts": {
+        "route": "FROM /messages/modules/FloorCamMod/outputs/P0_Alerts INTO $upstream",
+        "priority": 0,
+        "timeToLiveSecs": 86400
+      },
+      "secCamData": {
+        "route": "FROM /messages/modules/SecCamMod/outputs/P1_Data INTO $upstream",
+        "priority": 1,
+        "timeToLiveSecs": 1800
+      },
+      "upstream": "FROM /messages/* INTO $upstream"
+    },
+    "storeAndForwardConfiguration": {
+      "timeToLiveSecs": 7200
+    }
+  }
+}
+</pre>
+
+In this case, the upstream route is a “catch-all” to handles messages that are not explicitly routed with a priority.  In this case, some messages will now have two different routes to the same endpoint.  To preserve the design intent, Edge Hub will always use the route that yields the highest priority.
+
+### Components
+The message forwarding stack in Edge Hub is currently abstracted into the following layers:
+1. Protocol heads – receive messages from child modules and leaf devices, and translate the protocol-specific payloads into our own in-memory representation.
+2. Routing – takes an incoming message, matches the message source and payload to determine the destination endpoint as specified in the deployment manifest.
+3. Message queue/storage – takes a routed message and enqueues it into the destination endpoint’s sequential store, then asynchronously pump messages out of the queue to be sent as connectivity allows.  This layer also deals with message TTL and cleans up anything past the expiration timestamp.
+4. Endpoint – an interface backed by either a module endpoint, or the upstream cloud endpoint.  This layer utilizes with the device and module SDKs to actually send a message over the wire.
+
+![](./images/route_priority_ttl/legacy_routing_architecture.jpg)
+
+# Design decisions
+As mentioned above, the new asks from customer scenarios effectively boils down to handling when messages are sent.  For prioritized messages, it means some messages are sent earlier than others.  For route specific TTL, it means some messages might never get sent.  Both behaviors are best dealt with at the message queue/storage layer where our queue management can be made smarter.
```

Comment:
As mentioned above, the new asks from customer scenarios effectively boils down to handling when messages are sent --> Priorities and TTL are handled when messages are sent. ?